File: src/dataset/Java/isogram_exercism-java-uzilan_robiworks/openAI/ChatGPT4_isogram_v1.java
import java.util.HashSet;

class IsogramChecker {

    boolean isIsogram(String phrase) {
        phrase = phrase.toLowerCase();
        HashSet<Character> seen = new HashSet<>();
        for (char c : phrase.toCharArray()) {
            if (c != ' ' && c != '-') {
                if (!seen.add(c)) {
                    return false;
                }
            }
        }
        return true;
    }

}

File: src/dataset/typescript/isogram_Exercism-typescript-FilipeCerejo/openAI/ChatGPT4_isogram_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isIsogram = isIsogram;
function isIsogram(phrase) {
    const seen = new Set();
    for (const char of phrase.toLowerCase()) {
        if (/\w/.test(char)) {
            if (seen.has(char)) {
                return false;
            }
            seen.add(char);
        }
    }
    return true;
}
